chains/solana/contracts/programs/ccip-router/src/instructions/v1/token_admin_registry.rs

```diff
@@ -53,20 +60,91 @@ pub fn register_token_admin_registry_via_owner(
 }
 
 pub fn set_pool(
-    ctx: Context<ModifyTokenAdminRegistry>,
+    ctx: Context<SetPoolTokenAdminRegistry>,
     mint: Pubkey,
-    pool_lookup_table: Pubkey,
+    writable_indexes: Vec<u8>,
 ) -> Result<()> {
+    // set new lookup table
     let token_admin_registry = &mut ctx.accounts.token_admin_registry;
     let previous_pool = token_admin_registry.lookup_table;
-    token_admin_registry.lookup_table = pool_lookup_table;
+    let new_pool = ctx.accounts.pool_lookuptable.key();
+    token_admin_registry.lookup_table = new_pool;
 
-    // TODO: Validate here that the lookup table has everything
+    // set writable indexes
```

Comment:
What is this doing? Any of the accounts passed can be writable? Like the regular accounts and the additionals?

Reply:
correct, since these accounts are token pool specific, we need a way to know how to build the transaction offchain

this way provides an onchain record that admins can use to set which token pool accounts are writable